stages/org.osbuild.users

```diff
@@ -126,6 +142,8 @@ def usermod(root, name, gid=None, groups=None, description=None, home=None, shel
         arguments += ["--shell", shell]
     if password is not None:
         arguments += ["--password", password]
+    if expiredate is not None:
+        arguments += ["--expiredate", expiredate]
```

Comment:
Let's see what the tests think but I *think* this needs to be `str(expiredate)` if it comes out as an integer.

Reply:
This is an excellent catch Simon, sadly the tests will not catch this because they mock too much. 

Here is a quick draft of a test that ensures useradd can be called:
[0001-stage-add-user-test-that-ensures-the-binary-can-be-c.patch.txt](https://github.com/osbuild/osbuild/files/14556518/0001-stage-add-user-test-that-ensures-the-binary-can-be-c.patch.txt)
but I think we should do better and extend mock_command to be a bit smarter:
[0002-testutil-tweak-mock_command-to-write-a-call_log.patch.txt](https://github.com/osbuild/osbuild/files/14557009/0002-testutil-tweak-mock_command-to-write-a-call_log.patch.txt)
and then we can do:
```python
diff --git a/stages/test/test_users.py b/stages/test/test_users.py
index 1955391d..39d6441a 100644
--- a/stages/test/test_users.py
+++ b/stages/test/test_users.py
@@ -37,8 +37,8 @@ def test_users_happy(mocked_run, tmp_path, stage_module, user_opts, expected_arg
 
 
 @pytest.mark.parametrize("user_opts,expected_args", TEST_CASES)
-def test_users_can_be_called(tmp_path, stage_module, user_opts, expected_args):
-    with mock_command("chroot", "#!/bin/sh"):
+def test_users_mock_bin(tmp_path, stage_module, user_opts, expected_args):
+    with mock_command("chroot", "") as mock_chroot:
         make_fake_tree(tmp_path, {
             "/etc/passwd": "",
         })
@@ -51,4 +51,4 @@ def test_users_can_be_called(tmp_path, stage_module, user_opts, expected_args):
         options["users"]["foo"].update(user_opts)
 
         stage_module.main(tmp_path, options)
-
+        assert "\n".join(expected_args) in mock_chroot.read_text(encoding="utf8")
```
but I'm fine landing this in a followup especially since the mock_command patch could be improved further, the API is really a bit too basic.

I.e. to unblock this I think just fixing the str() and maybe adding the first diff should be enough :)

Reply:
Done!